### PR TITLE
Adding in Mapview

### PR DESCRIPTION
**Pull request by Gmiller290488**

### CODE REVIEW

File: iosHackthonApp/iosHackthonApp/Meal List Screen/MapView/MapView.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+struct MapView: UIViewRepresentable {
+
+    var annotations: [MKPointAnnotation]
+    @Binding var selectedPlace: MKPointAnnotation?
+
+    func makeUIView(context: Context) -> MKMapView {
+        let mapView = MKMapView()
+        mapView.delegate = context.coordinator
+        mapView.showsUserLocation = true
+        mapView.region = MKCoordinateRegion(center: mapView.userLocation.coordinate, latitudinalMeters: 1000, longitudinalMeters: 1000)
```
Comment (kirstybutler) [inline iosHackthonApp/iosHackthonApp/Meal List Screen/MapView/MapView.swift:21]: is a `region` similar to geofence?

Comment (Gmiller290488) [reply iosHackthonApp/iosHackthonApp/Meal List Screen/MapView/MapView.swift:21]: I don't know, I only heard geofence for the first time when you mentioned it 😄 
I would guess so. It's specifying an area around the user to zone in on

File: iosHackthonApp/iosHackthonApp/Meal List Screen/MealListView.swift [hunk 2]
```diff
@@ -7,52 +7,61 @@ struct MealListView: View {
...
+
+            if viewModel.showMap == 0 {
+                ScrollView {
+                    LazyVStack {
+                        ForEach((0 ..< viewModel.meals.count), id: \.self) {
+                            MealRow(listViewModel: viewModel, rowViewModel: MealRowViewModel(mealWithDistance: viewModel.meals[$0]))
+                                .padding()
+                        }
+                    }
+
```
Comment (kirstybutler) [inline iosHackthonApp/iosHackthonApp/Meal List Screen/MealListView.swift]: space?

File: iosHackthonApp/iosHackthonApp/Meal List Screen/MealListViewModel.swift [hunk 3]
```diff
@@ -34,10 +38,29 @@ final class MealListViewModel: ObservableObject {
...
                 .sorted(by: { $0.distance < $1.distance })
+
+            for meal in meals {
+                let mapAnnotation = MKPointAnnotation()
+                let location = CLLocationCoordinate2D(latitude: Double(meal.locationLat), longitude: Double(meal.locationLong))
+                mapAnnotation.coordinate = location
+                mapAnnotation.title = meal.name
+                mapAnnotation.subtitle = self.getQuantityText(Int(meal.quantity))
+                self.locations.append(mapAnnotation)
+
```
Comment (kirstybutler) [inline iosHackthonApp/iosHackthonApp/Meal List Screen/MealListViewModel.swift]: space?

File: iosHackthonApp/iosHackthonApp/Meal List Screen/MealListViewModel.swift [hunk 4]
```diff
@@ -34,10 +38,29 @@ final class MealListViewModel: ObservableObject {
...
+            return "\(quantity) \(Strings.MealListScreen.portions)"
+        } else if quantity == 1 {
+            return "\(quantity) \(Strings.MealListScreen.portion)"
+        }
+        return "\(Strings.MealListScreen.Map.reserved)"
+    }
+
+    private func mealNotExpired(_ expiry: String) -> Bool {
         Date() < Date.dateFrom(dateString: expiry)
     }
```
Comment (dsmith42) [inline iosHackthonApp/iosHackthonApp/Meal List Screen/MealListViewModel.swift:65]: Nice idea here, I'll port the quantityText over to AND

File: iosHackthonApp/iosHackthonApp/Resources/Colors/ColorManager.swift [hunk 5]
```diff
@@ -10,11 +10,10 @@ import SwiftUI
...
-	static let appPrimary = Color("appPrimary")
-	static let appSecondary = Color("appSecondary")
-	static let red = Color.red
-	static let blue = Color.blue
-	static let gray = Color.gray
+    static let red = Color.red
+    static let blue = Color.blue
+    static let gray = Color.gray
     static let appGreen = Color("appGreen")
+    static let appDarkGreen = Color("appDarkGreen")
```
Comment (kirstybutler) [inline iosHackthonApp/iosHackthonApp/Resources/Colors/ColorManager.swift]: Could the green and dark green be the primary and secondary?

Comment (Gmiller290488) [reply iosHackthonApp/iosHackthonApp/Resources/Colors/ColorManager.swift]: Sure.
That would also make more sense if we were to add Dark mode colours too - I'll change it

**Review by kirstybutler — APPROVED** [review]

**Review by dsmith42 — APPROVED** [review]
Looks good

**Review by raynelle03 — APPROVED** [review]

File: shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt [hunk 6]
```diff
@@ -42,9 +42,9 @@ class MealApi {
 
     companion object {
         // iOS can use this endpoint
-//        private const val MEALS_ENDPOINT = "http://localhost:3000/Meals"
+        private const val MEALS_ENDPOINT = "http://localhost:3000/Meals"
 
         // Android needs to switch to this endpoint
-        private const val MEALS_ENDPOINT = "http://10.0.2.2:3000/meals"
+//        private const val MEALS_ENDPOINT = "http://10.0.2.2:3000/meals"
```
Comment (Gmiller290488) [inline shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt:48]: This pickled my brain - thought I had broke something when nothing was showing 😬 
I'm gonna fix this after the demo to be a switch on platform type

Comment (kirstybutler) [reply shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt:48]: Nice idea

**Pull request merged**